### ISSUE TITLE
Error processing  enum type

### ISSUE THREAD

**Issue by akiarostami** [ciscoheat/sveltekit-superforms#5]
Hello, 

I'm not sure if I'm doing something wrong, or there's a bug, but I can't include enum in my schema. This single like makes the code silently break and nothing gets passed to the form object on the server on submit:

`gender: z.enum(["male", "female", "other"]).nullish()`

Am I using this wrong?

**Comment by ciscoheat** [comment]
Hi, Zod enums doesn't have an implicit default value so that should throw an exception, but now when it's `nullish`, it should be set to null as default. I'll look into it asap, thanks for the report!

**Comment by akiarostami** [comment]
Btw, I love how clean my code has become after started using this. Thank you!

**Comment by ciscoheat** [comment]
That makes me so glad to hear. :) So I've made a little test:

```ts
  const schema = z.object({
    gender: z.enum(['male', 'female', 'other']).nullish()
  });

  const form = await superValidate(null, schema);
  console.log(form);
```

Which gives the correct output for me:

```ts
{
  valid: false,
  errors: {},
  data: { gender: null },
  empty: true,
  message: null
}
```

What do you get when doing this?

**Comment by akiarostami** [comment]
This part was OK. Sorry I wasn't clear in my original post. 

This was the problem: when I submitted the form, values for all form entries were null (I have several fields). After I commented out enum, everything started working.

**Comment by ciscoheat** [comment]
Sorry, my fault, I didn't see that it was about submitting! Now I've done the same test when submitting, and I do get an error when setting a value for the enum field:

```
Error: Unsupported Zod default type: ZodEnum
```

When it is null, everything works though, all other form entries have their respective values. Could it be that the exception was thrown for you as well?

I've published version `0.5.8-dev.2` which fixes the above, can you update to that and see if it works?

**Comment by ciscoheat** [comment]
Sorry, closed this by mistake in the commit log. But the enum problem is patched in version 0.5.9, can you try that one out?

**Comment by akiarostami** [comment]
This seems to be working. Thank you! I'm closing this now.
